src/data/set/basic.lean

```diff
@@ -1687,18 +1679,14 @@ end image
 
 /-! ### Subsingleton -/
 
-/-- A set `s` is a `subsingleton`, if it has at most one element. -/
+/-- A set `s` is a `subsingleton` if it has at most one element. -/
 protected def subsingleton (s : set α) : Prop :=
 ∀ ⦃x⦄ (hx : x ∈ s) ⦃y⦄ (hy : y ∈ s), x = y
 
-lemma subsingleton.mono (ht : t.subsingleton) (hst : s ⊆ t) : s.subsingleton :=
```

Comment:
I think these changes to `subsingleton` should be separated into a smaller PR. They're good as far as I can tell, but they're independent of having the `set.nontrivial` predicate.

Reply:
I was worried you might say that (I am not sure how best now to separate them off).

Reply:
In particular it was a right pain to find all the files where `subsingleton.mono` was used.

Reply:
Well, it should be as simple as just doing this exact PR but without the new `nontrivial` additions, right?

Reply:
I suppose so, I shall just have to do it manually.

Reply:
#15886 now does this.